cmd/vulcan-exposed-ssh/main.go

```diff
@@ -211,62 +224,50 @@ func (r *runner) gradeVuln() ([]report.Vulnerability, error) {
 		// Evaluate recommendations and map them to vulnerabilities
 		for _, j := range i.Compliance.Recommendations {
 			if strings.Contains(j, kexRecommendationPattern) {
-				tmp, ok := vulnDb["weakKexConfigVuln"]
-				if !ok {
-					tmp = weakKexConfigVuln
-				}
-				tmp.Recommendations = append(tmp.Recommendations, j)
-				tmp.Details += fmt.Sprintf("* Affected port: %v\n", i.Port)
-				vulnDb["weakKexConfigVuln"] = tmp
+				v := weakKexConfigVuln
+				v.AffectedResource = fmt.Sprintf("%d/%s", i.Port, "ssh")
+				v.Recommendations = append(v.Recommendations, j)
+				v.Details += fmt.Sprintf("* Affected port: %v\n", i.Port)
+				v.Fingerprint = helpers.ComputeFingerprint(v.Summary, v.Score)
+				vulnArray = append(vulnArray, v)
 			} else if strings.Contains(j, ciphersRecommendationPattern) {
-				tmp, ok := vulnDb["weakCiphersConfigVuln"]
-				if !ok {
-					tmp = weakCiphersConfigVuln
-				}
-				tmp.Recommendations = append(tmp.Recommendations, j)
-				tmp.Details += fmt.Sprintf("* Affected port: %v\n", i.Port)
-				vulnDb["weakCiphersConfigVuln"] = tmp
+				v := weakCiphersConfigVuln
+				v.Recommendations = append(v.Recommendations, j)
+				v.Details += fmt.Sprintf("* Affected port: %v\n", i.Port)
+				v.AffectedResource = fmt.Sprintf("%d/%s", i.Port, "ssh")
+				v.Fingerprint = helpers.ComputeFingerprint(v.Summary, v.Score)
+				vulnArray = append(vulnArray, v)
 			} else if strings.Contains(j, macsRecommendationPattern) {
-				tmp, ok := vulnDb["weakMACsConfigVuln"]
-				if !ok {
-					tmp = weakMACsConfigVuln
-				}
-				tmp.Recommendations = append(tmp.Recommendations, j)
-				tmp.Details += fmt.Sprintf("* Affected port: %v\n", i.Port)
-				vulnDb["weakMACsConfigVuln"] = tmp
+				v := weakMACsConfigVuln
+				v.Recommendations = append(v.Recommendations, j)
+				v.Details += fmt.Sprintf("* Affected port: %v\n", i.Port)
+				v.AffectedResource = fmt.Sprintf("%d/%s", i.Port, "ssh")
+				v.Fingerprint = helpers.ComputeFingerprint(v.Summary, v.Score)
```

Comment:
If the summary changes, the entire issue will change so it's not useful for the fingerprint. I'd rather use or empty args, or the `ServerBanner` value. WDYT?